art/attacks/evasion/auto_projected_gradient_descent.py

```diff
@@ -323,9 +280,15 @@ def __call__(self, y_pred, y_true):  # type: ignore
                             z_i = torch.diagonal(z_i)
                             z_y = torch.diagonal(z_y)
 
-                            dlr = -(z_y - z_i) / (z_1 - z_3)
-
-                            return torch.mean(dlr.float())
+                            # modification for image-wise stepsize update
+                            dlr = (-(z_y - z_i) / (z_1 - z_3)).float()
+                            if self.reduction == "mean":
+                                return dlr.mean()
+                            if self.reduction == "sum":
+                                return dlr.sum()
+                            if self.reduction == "none":
+                                return dlr
+                            raise NotImplementedError()
 
                     self._loss_object = DifferenceLogitsRatioPyTorch()
 
```

Comment:
Please change line 297 from `loss=self._loss_object,` to `loss=_loss_object_pt,`